src/main/java/io/vertx/core/impl/launcher/commands/ExecUtils.java

```diff
@@ -0,0 +1,95 @@
+/*
+ *  Copyright (c) 2011-2015 The original author or authors
+ *  ------------------------------------------------------
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *       The Eclipse Public License is available at
+ *       http://www.eclipse.org/legal/epl-v10.html
+ *
+ *       The Apache License v2.0 is available at
+ *       http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.core.impl.launcher.commands;
+
+import java.util.List;
+
+/**
+ * A couple of utility methods easing process creation.
+ *
+ * @author Clement Escoffier <clement@apache.org>
+ */
+public class ExecUtils {
+
+  private static final String SINGLE_QUOTE = "\'";
+  private static final String DOUBLE_QUOTE = "\"";
+
+  /**
+   * The {@code os.name} property is mandatory (from the Java Virtual Machine specification).
+   */
+  private static String osName = System.getProperty("os.name").toLowerCase();
+
+  /**
+   * Puts quotes around the given String if necessary.
+   * <p>
+   * If the argument doesn't include spaces or quotes, return it as is. If it
+   * contains double quotes, use single quotes - else surround the argument by
+   * double quotes.
+   * </p>
+   *
+   * @param argument the argument to be quoted
+   * @return the quoted argument
+   * @throws IllegalArgumentException If argument contains both types of quotes
+   */
+  public static String quoteArgument(final String argument) {
+
+    String cleanedArgument = argument.trim();
+
+    // strip the quotes from both ends
```

Comment:
wouldn't it be better to strip both ends at the same time and check the quote is the same, instead of cleaning each end separately ?